File: frontend/src/mateu/shared/apiClients/dtos/Action.ts
import {ActionType} from "./ActionType";
import ConfirmationTexts from "./ConfirmationTexts";
import {ActionTarget} from "./ActionTarget";
import {ActionPosition} from "./ActionPosition";

export default interface Action {

    id: string
    icon: string
    caption: string
    type: ActionType
    visible: boolean
    validationRequired: boolean
    confirmationRequired: boolean
    rowsSelectedRequired: boolean
    confirmationTexts: ConfirmationTexts
    target: ActionTarget
    modalStyle: string
    customEvent: string
    href: string
    runOnEnter: boolean
    position: ActionPosition
    timeoutMillis: number

}

File: backend/dtos/src/main/java/io/mateu/remote/dtos/Action.java
package io.mateu.remote.dtos;

import lombok.*;

@Data
@Builder
@NoArgsConstructor(access = AccessLevel.PACKAGE)
@AllArgsConstructor(access = AccessLevel.PACKAGE)
public class Action {

  private String id;

  private String caption;

  private ActionType type;

  private boolean visible;

  private boolean validationRequired;

  private boolean confirmationRequired;

  private boolean rowsSelectedRequired;

  private ConfirmationTexts confirmationTexts;

  private ActionTarget target;

  private String modalStyle;

  private String customEvent;

  private String href;

}
